### PR TITLE
Add missing gnirs enums

### PR DESCRIPTION
**Pull request by cquiroz**
Somehow we missed porting these enums from gsp-core

### CODE REVIEW

File: modules/core/shared/src/main/scala/lucuma/core/enum/GnirsPrism.scala [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+package lucuma.core
+package enum
+
+import cats.instances.string._
+import cats.syntax.eq._
+import lucuma.core.util.Enumerated
+
+/**
+ * Enumerated type for GNIRS Prism Turret.
+ * @group Enumerations (Generated)
```
Comment (swalker2m) [inline modules/core/shared/src/main/scala/lucuma/core/enum/GnirsPrism.scala:13]: Are these still generated?  I thought we abandoned that idea.

File: modules/core/shared/src/main/scala/lucuma/core/enum/GnirsPrism.scala [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  val tag: String,
+  val shortName: String,
+  val longName: String
+) extends Product with Serializable {
+  type Self = this.type
+}
+
+object GnirsPrism {
+
+  type Aux[A] = GnirsPrism { type Self = A }
```
Comment (swalker2m) [inline modules/core/shared/src/main/scala/lucuma/core/enum/GnirsPrism.scala:25]: Remind me what this is about?

File: modules/core/shared/src/main/scala/lucuma/core/enum/GnirsReadMode.scala [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+/**
+ * Enumerated type for GNIRS Read Mode.
+ * @group Enumerations (Generated)
+ */
+sealed abstract class GnirsReadMode(
+  val tag: String,
+  val shortName: String,
+  val longName: String,
+  val count: Int,
+  val minimumExposureTime: Int,
```
Comment (swalker2m) [inline modules/core/shared/src/main/scala/lucuma/core/enum/GnirsReadMode.scala:20]: Can this be a `Duratation` or something now?

File: modules/core/shared/src/main/scala/lucuma/core/enum/GnirsReadMode.scala [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+  val readNoise: Int,
+  val readNoiseLow: Int
```
Comment (swalker2m) [inline modules/core/shared/src/main/scala/lucuma/core/enum/GnirsReadMode.scala:22]: Should we represent these with their units now?

File: modules/core/shared/src/main/scala/lucuma/core/enum/GnirsPrism.scala (hunk 2 above)
Comment (tpolecat) [reply modules/core/shared/src/main/scala/lucuma/core/enum/GnirsPrism.scala:25]: Not needed. It was from that thing where I was experimenting with indexing thpes on the singleton types of enum values, but it was too complicated.

File: modules/core/shared/src/main/scala/lucuma/core/enum/GnirsPrism.scala (hunk 1 above)
Comment (tpolecat) [reply modules/core/shared/src/main/scala/lucuma/core/enum/GnirsPrism.scala:13]: Not generated anymore. I think we agreed that the Scala enum is truth and database apps need to do a some kind of startup check to ensure consistency (not implemented).

**Comment by cquiroz** [comment]
These were generated ages ago and we copied them to Lucuma core, for some
Reason we didn't copy thes two and we need them in the seqexec 
I copied them only changin the package name

**Review by swalker2m — APPROVED** [review]
> These were generated ages ago and we copied them to Lucuma core

I see.  We probably need to go through all of them then at some point and clean them up.

**Pull request merged**